backend/webserver/views.py

```diff
@@ -70,6 +72,109 @@ def post(self, request):
         return Response({'message': 'Successfully Logged out'}, status=status.HTTP_200_OK)
 
 
+class PostView(APIView):
+    authentication_classes = [BasicAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_author(self,pk):
+        author = get_object_or_404(Author,pk=pk)
+        return author
+    def get_post(self,post_id,author_id):
+        post = get_object_or_404(Post,id=post_id,author=author_id)
+        return post
+    
+    def get(self,request,pk,post_id, *args, **kwargs):
+        author =self.get_author(pk)
+        post = self.get_post(post_id,author.id)
+        if post.visibility == "PUBLIC":
+            serializer = PostSerializer(post, context={'request': request})
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response({'message': 'requested post is not public'}, status=status.HTTP_400_BAD_REQUEST)
+            
+
+    def post(self, request, pk,post_id, *args, **kwargs):
+        author = self.get_author(pk)
+        post = self.get_post(post_id,author.id)
+
+        if post.visibility == "PUBLIC":   
+            if author.id == request.user.id:
+                post.edited_at = datetime.datetime.utcnow().replace(tzinfo=utc)
+                serializer = UpdatePostSerializer(instance=post,data=request.data,partial=True,context={'request': request})
+                if serializer.is_valid():
+                    serializer.save()
```

Comment:
```suggestion
                serializer = UpdatePostSerializer(instance=post,data=request.data,partial=True,context={'request': request})
                if serializer.is_valid():
                    serializer.save(edited_at=datetime.datetime.utcnow().replace(tzinfo=utc))
```
If serializer is invalid, post.edited_at will still change. Do we want that?

The syntax I am suggesting is taken [from here](https://www.django-rest-framework.org/api-guide/serializers/#passing-additional-attributes-to-save).